### ISSUE TITLE
Reduce number of status changes written to status bar

### ISSUE THREAD

**Issue by lardconcepts** [ableplayer/ableplayer#374]
AblePlayer 4.0.0.8 (dev)
NVDA 2019.1.1
Chrome 76.0.3809.87 (Official Build) beta (64-bit)

When the next track starts playing in Chrome, NVDA says:

"Selected track(trackname) paused, playing", but only on Chrome. Firefox is OK.

Is there any way to suppress this?

**Comment by terrill** [comment]
@lardconcepts currently, Able Player has up to six items tagged as ARIA live regions. Screen readers are expected to announce the content of these regions as it changes. Four of these are "assertive" (screen readers announce the change immediately), and two are "polite" (screen readers announce the change later, when there's a break in the action). The live regions are: 

**Assertive Live Regions**
- "Selected Track" (so users will know what's playing when the track changes) 
- Volume slider (announces X% after user changes volume) 
- Speed indicator on status bar (announces the new speed after a speed change) 
- Audio description text (if available) 

**Polite Live Regions** 
Seekbar position (announces new location, e.g., "1 minute 19 seconds", if user moves the seekbar)
Status (e.g., "Playing", "Paused", "Buffering") 

What you're observing in Chrome is NVDA announcing the title of the new track (which is arguably important information). The arguably less important information is the current status (e.g. paused, playing, buffering) but in discussing this with users we've found disagreement as to whether they feel this is important information. If a new track starts off silent, how do they know for sure whether it's playing, unless the "Playing" or "Paused" status is made accessible, as it is to sighted users?  That said, some users prefer having less audible clutter. Our compromise was to set the status changes to aria-live="polite", so screen readers should be more conservative about when and whether they announce this information to users, but all screen reader/browser combinations are a little different. 

One solution would be to add a delay before updating the status, particularly in playlists. Typically the status changed to "Paused" after a track ends, while the new media is being loaded, then it changes back to "Play". We could write this change to the status bar only if it takes greater than a predefined number of seconds to load the new media. I'll look into that.

**Comment by lardconcepts** [comment]
That delay idea would be a perfect solution. Thanks

**Comment by terrill** [comment]
This has been addressed in 1da51f5ebc9739454045ab7d202601102bf2814c

I added two thresholds: 

First, if the player is swapping the media source (e.g., from one playlist track to another or from non-described to described version), a change in status must continue for at least 2 seconds before the new status is written to the status bar. If a message continues longer than that (e.g., "Buffering") users should know about it.  
 
Second, for all other changes in status, the threshold is shorter (250 ms). This is for instance like when the user presses "Play", and should arguably receive prompt feedback that the media is "Playing" (especially needed for media that starts quiet).
